Sources/HTTP/Request/Request.swift

```diff
@@ -0,0 +1,66 @@
+import Core
+import URI
+
+public final class Request: Message {
+    public var method: Method
+    public var uri: URI
+    public var version: Version
```

Comment:
Having these variables constant (and especially those on the URI) have made using the Engine APIs difficult in the past. I don't see any reason why we need to prevent users from being able to change these values.